rfcs/0006-runtime-references-interface.md

```diff
@@ -0,0 +1,77 @@
+---
+feature: runtime-references-interface
+start-date: 2017-04-01
+author: Shea Levy
+co-authors: (find a buddy later to help our with the RFC)
+related-issues: https://github.com/NixOS/rfcs/pull/6
+---
+
+# Summary
+[summary]: #summary
+
+Currently, nix itself scans derivation outputs for the hashes of the
+(transitive closure of) the inputs of the build and registers those it
+finds as references. This feature would provide an optional interface
+for builds to specify their own runtime references, and possibly for
+nix to tell the build what its build time inputs are, bypassing nix's
+own hash scanning.
+
+# Motivation
+[motivation]: #motivation
+
+The existing hash scanning works very well for most use cases. There
+are times, however, where we have to hack around the constraints it
+imposes (e.g. `nukeReferences`, not storing the configure flags used
+to build gcc, storing the path of a file in a comment to ensure the
+build depends on it, etc.). Moreover, every time we need to tweak the
+logic we need a new nix release. Finally, recursive nix will involve a
+more complex reference checking story. Having an interface to
+explicitly specify runtime references would improve all of these
+issues.
+
+Nix will need to provide the build time references to the build to
+enable us to move reference scanning to stdenv and to allow for setups
+that use the reference scan results as a base and then modify that set
+accordingly.
+
+# Detailed design
+[design]: #detailed-design
+
+Nix will put all the build time references (or requisites?) into
+`$NIX_BUILD_TOP/build-time-references`. After the build completes, nix
+will check `$NIX_BUILD_TOP/runtime-references` and, if it exists, skip
+its builtin hash scanning and just use the file. Both files are
+newline-separated lists of store paths. Parse errors in the
+`runtime-references` file, or non-existent store paths, fail the build.
+
+Alternatively, since recursive nix will require a more complex
+interface with the host nix process anyway, we can just expose a unix
+domain socket (possibly just a socketpair) to the build to query the
```

Comment:
I don't understand the question. There is plenty of nix daemon interaction when using nix as a package manager, except in single user mode.